sklearn/preprocessing/tests/test_data.py

```diff
@@ -1845,136 +1787,73 @@ def test_normalizer_l1():
-        normalizer = Normalizer(norm="l1", copy=True)
-        X_norm = normalizer.transform(X)
-        assert X_norm is not X
-        X_norm1 = toarray(X_norm)
-
-        normalizer = Normalizer(norm="l1", copy=False)
-        X_norm = normalizer.transform(X)
-        assert X_norm is X
-        X_norm2 = toarray(X_norm)
-
-        for X_norm in (X_norm1, X_norm2):
-            row_sums = np.abs(X_norm).sum(axis=1)
-            for i in range(3):
-                assert_almost_equal(row_sums[i], 1.0)
-            assert_almost_equal(row_sums[3], 0.0)
-
-    # check input for which copy=False won't prevent a copy
-    for init in (sparse.coo_matrix, sparse.csc_matrix, sparse.lil_matrix):
-        X = init(X_dense)
-        X_norm = normalizer = Normalizer(norm="l2", copy=False).transform(X)
-
-        assert X_norm is not X
-        assert sparse.issparse(X_norm) and X_norm.format == "csr"
-
-        X_norm = toarray(X_norm)
-        for i in range(3):
-            assert_almost_equal(row_sums[i], 1.0)
-        assert_almost_equal(la.norm(X_norm[3]), 0.0)
-
-
-def test_normalizer_l2():
-    rng = np.random.RandomState(0)
-    X_dense = rng.randn(4, 5)
-    X_sparse_unpruned = sparse.csr_matrix(X_dense)
-
-    # set the row number 3 to zero
-    X_dense[3, :] = 0.0
-
-    # set the row number 3 to zero without pruning (can happen in real life)
-    indptr_3 = X_sparse_unpruned.indptr[3]
-    indptr_4 = X_sparse_unpruned.indptr[4]
-    X_sparse_unpruned.data[indptr_3:indptr_4] = 0.0
-
-    # build the pruned variant using the regular constructor
-    X_sparse_pruned = sparse.csr_matrix(X_dense)
-
-    # check inputs that support the no-copy optim
-    for X in (X_dense, X_sparse_pruned, X_sparse_unpruned):
-        normalizer = Normalizer(norm="l2", copy=True)
+        normalizer = Normalizer(norm=norm, copy=True)
         X_norm1 = normalizer.transform(X)
         assert X_norm1 is not X
         X_norm1 = toarray(X_norm1)
 
-        normalizer = Normalizer(norm="l2", copy=False)
+        normalizer = Normalizer(norm=norm, copy=False)
         X_norm2 = normalizer.transform(X)
         assert X_norm2 is X
         X_norm2 = toarray(X_norm2)
 
         for X_norm in (X_norm1, X_norm2):
-            for i in range(3):
-                assert_almost_equal(la.norm(X_norm[i]), 1.0)
-            assert_almost_equal(la.norm(X_norm[3]), 0.0)
-
-    # check input for which copy=False won't prevent a copy
-    for init in (sparse.coo_matrix, sparse.csc_matrix, sparse.lil_matrix):
-        X = init(X_dense)
-        X_norm = normalizer = Normalizer(norm="l2", copy=False).transform(X)
-
-        assert X_norm is not X
-        assert sparse.issparse(X_norm) and X_norm.format == "csr"
-
-        X_norm = toarray(X_norm)
-        for i in range(3):
-            assert_almost_equal(la.norm(X_norm[i]), 1.0)
-        assert_almost_equal(la.norm(X_norm[3]), 0.0)
-
-
-def test_normalizer_max():
+            if norm == "l1":
```

Comment:
Could you put the following check inside a function. It seems that we reuse-it below.

Reply:
Not sure if the modification is what you meant, please let me know if it is not.